onnxruntime/python/tools/transformers/models/llama/convert_to_onnx.py

```diff
@@ -0,0 +1,579 @@
+                        "weight": {"dtype": ["fp32"]},
+                        "activation": {"dtype": ["fp32"]},
+                    }
+                },
+            )
+
+            # Convert decoder_model.onnx to INT8
+            decoder_model_int8 = intel_quantization.fit(
+                decoder_model_fp32_path,
+                quantization_config,
+                calib_dataloader=QuantKVDataLoader(args),
+            )
+            load_external_data_for_model(
+                decoder_model_int8._model,
+                os.path.split(decoder_model_int8._model_path)[0],
+            )
+            save_onnx_model(
+                decoder_model_int8._model,
+                decoder_model_int8_path,
+                f"{args.model_name}_decoder_model_int8.onnx.data",
+            )
+            del decoder_model_int8
+
+            # Convert decoder_with_past_model.onnx to INT8
+            decoder_with_past_model_int8 = intel_quantization.fit(
+                decoder_with_past_model_fp32_path,
+                quantization_config,
+                calib_dataloader=QuantKVDataLoader(args, onnx_model_path=decoder_model_fp32_path),
+            )
+            load_external_data_for_model(
+                decoder_with_past_model_int8._model,
+                os.path.split(decoder_with_past_model_int8._model_path)[0],
+            )
+            save_onnx_model(
+                decoder_with_past_model_int8._model,
+                decoder_with_past_model_int8_path,
+                f"{args.model_name}_decoder_with_past_model_int8.onnx.data",
+            )
+            del decoder_with_past_model_int8
+
+            logger.info(f"Removing {args.nc_workspace}")
+            os.system(f"rm -R {args.nc_workspace}")
+
+        elif args.quantize_dynamic:
+            # Convert decoder_model.onnx to INT8
+            ort_quantization.quantize_dynamic(
+                decoder_model_fp32_path,
+                decoder_model_int8_path,
+                op_types_to_quantize=["MatMul", "Gemm", "Gather"]
+                if args.quantize_embedding_layer
+                else ["MatMul", "Gemm"],
+                per_channel=args.quantize_per_channel,
+                reduce_range=args.quantize_reduce_range,
+                use_external_data_format=True,
+                extra_options={"MatMulConstBOnly": True},
+            )
+
+            # Convert decoder_with_past_model.onnx to INT8
+            ort_quantization.quantize_dynamic(
+                decoder_with_past_model_fp32_path,
+                decoder_with_past_model_int8_path,
+                op_types_to_quantize=["MatMul", "Gemm", "Gather"]
+                if args.quantize_embedding_layer
+                else ["MatMul", "Gemm"],
+                per_channel=args.quantize_per_channel,
+                reduce_range=args.quantize_reduce_range,
+                use_external_data_format=True,
+                extra_options={"MatMulConstBOnly": True},
+            )
```

Comment:
what if neither --smooth_quant nor --quantize_dynamic is specified.

I suggest to merge them into one argument, like --quantization_method, choice is ["smooth_quant", "quantize_dynamic"], and default value is "smooth_quant". In the future, we will also need support int4 quantization methods.

Reply:
Added `--quantization_method` and deprecated `quantize_dynamic`. For INT4 quantization, we can use the `--precision` flag.

Reply:
Added `--quantization_method` and deprecated `quantize_dynamic`. For INT4 quantization, we can use the `--precision` flag and specify the method with `--quantization_method`.